src/sql/src/plan/transform_ast.rs

```diff
@@ -811,6 +811,11 @@ impl<'a> Desugarer<'a> {
                         }
                         *expr = new;
                     }
+                    _ if left.len() == 1 && right.len() == 1 => {
+                        let left = left.remove(0);
+                        let right = right.remove(0);
+                        *expr = left.binop(op.clone(), right);
+                    }
```

Comment:
It allows us to match PostgreSQL's behavior for comparing single-element row constructors when the operator is not `=`, `<`, `>`, `<=`, or `>=`:

```
benesch=# select row('a') like row('a%');
 ?column? 
----------
 t
(1 row)

benesch=# select row('a', 'A') like row('a%', 'A%');
ERROR:  could not determine interpretation of row comparison operator ~~
LINE 1: select row('a', 'A') like row('a%', 'A%');
                             ^
HINT:  Row comparison operators must be associated with btree operator families.
```

Because of the way we desugar `ANY` and `ALL` this is required to support `[NOT] [I]LIKE` as an `ANY`/`ALL` operator.